### PR TITLE
MM-47188 : Migrate "utils/message_html_to_component.jsx" to TypeScript

### PR DESCRIPTION
**Pull request by veronicadip**
I've read the Contribution Checklist, your blog post, and the repository documentation.

#### Summary
Migrated `message_html_to_component.jsx` and `message_html_to_component.test.jsx` to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/21163
https://mattermost.atlassian.net/browse/MM-47188

#### Release Note
```
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]
Nothing blocking, but a few potential improvements.

**Comment by lieut-data** [comment]
Apologies, I won't have the time to be able to review this one -- hoping @calebroseland can help (or delegate again ;P)

File: webapp/channels/src/types/external/html-to-react.d.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+declare module 'html-to-react' {
+    interface Node {
+        type: string;
+        name: string;
+        attribs: {
+            type: string;
+            checked: boolean;
+            'data-edited-post-id': string;
+            href: any;
+            mentionAttrib: string | undefined;
```
Comment (calebroseland) [inline webapp/channels/src/types/external/html-to-react.d.ts:18]: Are `data-edited-post-id` and `mentionAttrib` needed? Could they be covered by the index signature?

**Review by calebroseland — CHANGES_REQUESTED** [review]
Thanks for the contribution, @veronicadip! Seems like there are some mission properties on `Options`: `emoji`, `images`, `latex`, `inlinelatex`, & `markdown`—would you mind adding those, too?

**Comment by jespino** [comment]
This was handled internally. Anyway, thanks for the effort and time dedicated here @veronicadip! We really appreciate it.

**Pull request closed without merging**